subgraphs/arbitrum-one-bridge/src/handlers/arbitrum/transferIn.ts

```diff
@@ -0,0 +1,97 @@
+import { Address, Bytes, ethereum, log } from "@graphprotocol/graph-ts";
+import {
+  DepositFinalized,
+  TokenGateway,
+} from "../../../generated/ERC20Gateway/TokenGateway";
+import {
+  BridgePoolType,
+  CrosschainTokenType,
+} from "../../sdk/protocols/bridge/enums";
+import { SDK } from "../../sdk/protocols/bridge";
+import { networkToChainID } from "../../sdk/protocols/bridge/chainIds";
+import { arbSideConf, Pricer, TokenInit } from "../../common/utils";
+import { Network } from "../../sdk/util/constants";
+
+export function handleTransferIn3pGateway(event: DepositFinalized): void {
+  log.error("[3p Gateway] We are in transferIn3pGateway", []);
+
+  const l1Token = new ethereum.EventParam("l1token", event.parameters[0].value);
+  const _from = new ethereum.EventParam("_from", event.parameters[1].value);
+  const _to = new ethereum.EventParam("_to", event.parameters[2].value);
+  const _amount = new ethereum.EventParam("_amount", event.parameters[3].value);
+
+  const params: ethereum.EventParam[] = [l1Token, _from, _to, _amount];
+  const depositFinalized = new DepositFinalized(
+    event.address,
+    event.logIndex,
+    event.transactionLogIndex,
+    event.logType,
+    event.block,
+    event.transaction,
+    params,
+    event.receipt
+  );
+  handleTransferIn(depositFinalized);
+}
+
+export function handleTransferIn(event: DepositFinalized): void {
+  // -- SDK
+
+  const sdk = SDK.initialize(
+    arbSideConf,
+    new Pricer(event.block),
+    new TokenInit(),
+    event
+  );
+
+  // -- TOKENS
+
+  const gatewayContract = TokenGateway.bind(event.address);
+
+  const inputTokenAddress = event.params.l1Token;
+  const inputToken = sdk.Tokens.getOrCreateToken(inputTokenAddress);
+
+  let crossTokenAddress: Address;
+
+  const crossTokenAddressResult =
+    gatewayContract.try_calculateL2TokenAddress(inputTokenAddress);
+  if (crossTokenAddressResult.reverted) {
+    log.info("calculate cross token address call reverted", []);
+  } else {
+    crossTokenAddress = crossTokenAddressResult.value;
+  }
+
+  const crossToken = sdk.Tokens.getOrCreateCrosschainToken(
+    networkToChainID(Network.ARBITRUM_ONE),
+    crossTokenAddress!,
+    CrosschainTokenType.CANONICAL,
+    crossTokenAddress!
```

Comment:
Just noticing that this is not well documented, but the first address is the one of the crosschain token, and the second one is the address of the token on the current chain, or what you call `inputTokenAddress`.

Reply:
Ah, I was wondering about that but forgot diving into the underlying method. We should rename the params to be intuitive in that case (eg. inputTokenAddress, crossTokenAddress).

I'll go push the change out to pass the inputTokenAddress for the second one.